### PR TITLE
Add Unit Test:describe pod's node info when node is empty

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
If a pod's node name is empty, when to describe it, showing <none> instead.

**Special notes for your reviewer**:
Add unittest for #45788. 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CaoShuFeng** [comment]
Seems good to me.

File: pkg/printers/internalversion/describe_test.go [hunk 1]
```diff
@@ -94,6 +94,30 @@ func TestDescribePodNode(t *testing.T) {
...
+			HostIP: "",
+		},
+	})
+	c := &describeClient{T: t, Namespace: "foo", Interface: fake}
+	d := PodDescriber{c}
+	out, err := d.Describe("foo", "bar", printers.DescriberSettings{ShowEvents: true})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	if !strings.Contains(out, "<none>") {
```
Comment (eparis) [inline pkg/printers/internalversion/describe_test.go]: This seems inadequate. Wouldn't any such pods also have `Tolerations: <none>` ?  Seems you need to look for `Node: <none>`

**Comment by eparis** [comment]
/release-note-none
@k8s-bot ok to test

**Comment by zhangxiaoyu-zidif** [comment]
@eparis Sir, thank you for review. A pr #45572 causes me think if the node is empty, we need add some UT.
If it is not happened ever, I suggest to close the PR. : ) So far, I just pass UT locally. Not sure the fact...

**Comment by zhangxiaoyu-zidif** [comment]
@eparis Sir, actually the case that node being "" is often. Such as e2e test or other unusual cases. If we show the user "/", that seems to be odd.
For example:

'''shell
[root@pxy-vm:~/gopath/src/k8s.io/kubernetes]$ cluster/kubectl.sh get pods --all-namespaces=true  
NAMESPACE                    NAME           READY     STATUS    RESTARTS   AGE
e2e-tests-sched-pred-0g2pv   overcommit-0   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-1   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-2   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-3   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-4   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-5   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-6   0/1       Pending   0          3m
e2e-tests-sched-pred-0g2pv   overcommit-7   0/1       Pending   0          3m
 
[root@pxy-vm:~/gopath/src/k8s.io/kubernetes]$ cluster/kubectl.sh describe pods overcommit-0 --namespace=e2e-tests-sched-pred-0g2pv
Name:           overcommit-0
Namespace:      e2e-tests-sched-pred-0g2pv
Node:           /
Labels:         name=overcommit-0
                startPodsID=140616c1-501a-11e7-b144-002dc72800b3
Annotations:    <none>
Status:         Pending
IP:
Controllers:    <none>
Containers:
  overcommit-0:
    Image:      gcr.io/google_containers/pause-amd64:3.0
    Port:
    Limits:
      cpu:      500m
    Requests:
      cpu:              500m
    Environment:        <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-n97bl (ro)
'''

Hope it is convinced. thank you :)

**Comment by eparis** [comment]
Looking at your output we see both: `Controllers:    <none>` and `Environment:        <none>`. So this output would actually cause your UT to pass. It only looks for `<none>`. Instead your UT needs to check explicitly for `Node: <none>`. Does that make sense?

The Unit Test is very valuable. Thank you for writing it. But it doesn't seem to actually solve the problem we are trying to solve.

**Comment by zhangxiaoyu-zidif** [comment]
@eparis Thanks for your review, sir. I fix the output check. using "node:  <none>" to replace "<node>".
PTAL. Thanks again. :)

**Pull request closed without merging**